numpy/core/tests/test_multiarray.py

```diff
@@ -4887,6 +4887,56 @@ def test_locale(self):
             self.test_tofile_sep()
             self.test_tofile_format()
 
+    def test_deprecate_unparsable_file(self):
+        x = np.array([1.51, 2, 3.51, 4], dtype=float)
+        with open(self.filename, 'w') as f:
+            x.tofile(f, sep=',', format='%.2f')
+            f.write(',invalid')
+            f.close()
+
+        with assert_warns(DeprecationWarning):
+            res = np.fromfile(self.filename, sep=",")
+            assert_array_equal(res, x)
+
+        with assert_warns(DeprecationWarning):
+            # currently silently truncates:
+            res = np.fromfile(self.filename, sep=",", count=5)
+            assert_array_equal(res, x)
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("error", DeprecationWarning)
+            with assert_raises(DeprecationWarning):
+                np.fromfile(self.filename, sep=",")
```

Comment:
Consider putting these tests in test_deprecations.py, which combines the warning and error tests info a single helper

Reply:
Ah, true there is still stat one advantage of my funny file (the main reason for it used to be that testing deprecations was unreliable with pythons buggy warning management). But I guess it sometimes can be nice to have specific deprecation-only tests in one place.